toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1477,45 +1479,97 @@ func ConfigureUserStartupCommand(installChroot *safechroot.Chroot, username stri
 	return
 }
 
-func ProvisionUserSSHCerts(installChroot *safechroot.Chroot, username string, sshPubKeyPaths []string) (err error) {
-	var (
-		pubKeyData []string
-		exists     bool
-	)
-	const squashErrors = false
-	const authorizedKeysTempFilePerms = 0644
-	const authorizedKeysTempFile = "/tmp/authorized_keys"
-	const sshDirectoryPermission = "0700"
+func setupSSHEnvironment(authorizedKeysTempFilePerms os.FileMode, authorizedKeysTempFile string) (err error) {
+
+	// Check if the temporary authorized_keys file exists
+	exists, err := file.PathExists(authorizedKeysTempFile)
+	if err != nil {
+		return fmt.Errorf("error accessing %s file: %v", authorizedKeysTempFile, err)
+	}
+
+	if !exists {
+		// Create the file if it doesn't exist
+		err = file.Create(authorizedKeysTempFile, authorizedKeysTempFilePerms)
+		if err != nil {
+			return fmt.Errorf("failed to create %s file: %v", authorizedKeysTempFile, err)
+		}
+	} else {
+		// Truncate the file if it exists
+		err = os.Truncate(authorizedKeysTempFile, 0)
+		if err != nil {
+			return fmt.Errorf("failed to truncate %s file: %v", authorizedKeysTempFile, err)
+		}
+	}
+
+	return nil
+}
 
+func ProvisionUserSSHCertsWithPubKeys(username string, sshPubKeys []string) (err error) {
```

Comment:
You should allow the user to specify both `SSHPubKeyPaths` and `SSHPubKeys` and then concatenate them together. This will give the user extra flexibility. It would probably make the code in this file a little simpler, since `ProvisionUserSSHCertsWithPubKeys` and `ProvisionUserSSHCerts` will be combined into a single function and won't need to duplicate logic.